### PR TITLE
HPCC-28973 Document --main options

### PR DESCRIPTION
**Pull request by g-pan**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Completed unit testing: http://10.224.20.18/view/Docs-gp/job/Doc-Build-04-GPRepo2/

### CODE REVIEW

File: docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml [hunk 1]
```diff
@@ -277,6 +271,58 @@ ECLCC_PATH
               </tgroup>
             </informaltable></para>
         </sect3>
+
+        <sect3 id="CT-CLI-GIT-Support">
+          <title>Git Support</title>
+
+          <para>Starting with the HPCC Systems version 8.4, the
+          <emphasis>--main</emphasis> option has been extended to support
+          compiling a query directly from a git repository. When invoked, It
```
Comment (JamesDeFabia) [inline docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml:280]: capitalization: Git repository

File: docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml [hunk 2]
```diff
@@ -277,6 +271,58 @@ ECLCC_PATH
...
+          retrieves the ECL code specified from the git repository, compiles
```
Comment (JamesDeFabia) [inline docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml:281]: capitalization: Git repository

File: docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml [hunk 3]
```diff
@@ -277,6 +271,58 @@ ECLCC_PATH
...
+          the code and runs the query (on Thor). Only the checkout will be
```
Comment (JamesDeFabia) [inline docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml:282]: comma: the code, and runs the query

File: docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml [hunk 4]
```diff
@@ -277,6 +271,58 @@ ECLCC_PATH
...
+          line tools.</para>
+
+          <sect4 id="CT-Main-Addl-Examples">
+            <title>Additional Examples</title>
+
+            <para>The following examples all compile the attribute
+            <emphasis>demo.main</emphasis> from <emphasis>version3</emphasis>
+            branch of the github repository gituser/gch-ecldemo-d:</para>
+
+            <programlisting>eclcc --main demo.main@gituser/gch-ecldemo-d#version3mainrepoversion=version3
```
Comment (JamesDeFabia) [inline docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml:311]: Is this syntax correct? It looks odd.

File: docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml (hunk 3 above)
Comment (JamesDeFabia) [inline docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml:282]: is this only valid on Thor?

File: docs/EN_US/HPCCClientTools/CT_Mods/ECLCC.xml [hunk 5]
```diff
@@ -185,8 +185,9 @@
               <row>
                 <entry>-main <emphasis>&lt;ref&gt;</emphasis></entry>
 
-                <entry>Compile definition &lt;ref&gt; from the source
-                collection</entry>
+                <entry>Compile definition &lt;ref&gt; from the specified
+                source - note extended github repository support (see "Git
```
Comment (JamesDeFabia) [inline docs/EN_US/HPCCClientTools/CT_Mods/ECLCC.xml:189]: Is it only GitHub or does this work with any Git repository? If you are going to leave GitHub in the sentence, then it should use the capitalization they use--GitHub

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]
Comments inline

File: docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml (hunk 4 above)
Comment (g-pan) [reply docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml:311]: @ghalliday can confirm

File: docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml (hunk 3 above)
Comment (g-pan) [reply docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml:282]: I think so, @ghalliday can you please confirm?

File: docs/EN_US/HPCCClientTools/CT_Mods/ECLCC.xml (hunk 5 above)
Comment (g-pan) [reply docs/EN_US/HPCCClientTools/CT_Mods/ECLCC.xml:189]: I believe works with any Git - GitHub or GitLab, however perhaps @ghalliday can confirm?

File: docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml (hunk 4 above)
Comment (ghalliday) [reply docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml:311]: Either 
```
eclcc --main demo.main@gituser/gch-ecldemo-d#version3
```
or
```
eclcc --main demo.main@gituser/gch-ecldemo-d --mainrepoversion=version3
```

File: docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml (hunk 3 above)
Comment (ghalliday) [reply docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml:282]: No, it is valid on any target.

File: docs/EN_US/HPCCClientTools/CT_Mods/ECLCC.xml (hunk 5 above)
Comment (ghalliday) [reply docs/EN_US/HPCCClientTools/CT_Mods/ECLCC.xml:189]: It can work with github or gitlab, but it defaults to using a prefix that will use github

Comment (g-pan) [reply docs/EN_US/HPCCClientTools/CT_Mods/ECLCC.xml]: @JamesDeFabia note the <link> for the see also in the main option table entry does in fact generate working link : http://10.224.20.18/view/Docs-gp/job/DocBuild-01-GPRepo/65/artifact/build-docs/Release/docs/EN_US/TheECLIDEandHPCCClientTools_EN_US-9.0.7-closedown0.pdf#page=115

File: docs/EN_US/HPCCClientTools/CT_Mods/ECLCC.xml [hunk 6]
```diff
@@ -185,8 +185,10 @@
               <row>
                 <entry>-main <emphasis>&lt;ref&gt;</emphasis></entry>
```
Comment (JamesDeFabia) [inline docs/EN_US/HPCCClientTools/CT_Mods/ECLCC.xml:186]: should this be --main (two hyphens)?

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]
One comment inline

File: docs/EN_US/HPCCClientTools/CT_Mods/ECLCC.xml (hunk 6 above)
Comment (g-pan) [reply docs/EN_US/HPCCClientTools/CT_Mods/ECLCC.xml:186]: that is the way it appears in the existing doc, it was not changed

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]
ECLCC still has -main when I think it should be --main (line 186 of ECLCC.xml)

File: docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml [hunk 7]
```diff
@@ -277,6 +271,59 @@ ECLCC_PATH
...
+          <emphasis>--main</emphasis> option has been extended to support
+          compiling a query directly from a Git repository. When invoked, It
+          retrieves the ECL code specified from the Git repository, compiles
+          the code, and runs the query. Only the checkout will be done on the
+          remote ECLCCServer rather than on the client machine. Here is the
+          syntax to reference a repository location:</para>
+
+          <programlisting>&lt;protocol:&gt;//&lt;urn&gt;/&lt;user&gt;/&lt;repository&gt;#version </programlisting>
+
+          <para>While the <emphasis>#version</emphasis> component specifies
```
Comment (ghalliday) [inline docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml:288]: This sentence doesn't read clearly.  Possibly delete "specifies and" and replace "could" with "can"

**Review by ghalliday — COMMENTED** [review]
@g-pan one comment,and see response about "-main"

File: docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml [hunk 8]
```diff
@@ -277,6 +271,59 @@ ECLCC_PATH
...
+          <para>For example:</para>
+
+          <para><programlisting>ecl run thor --main demo.main@gituser/gch-ecldemo-d#version1 --server=...
+</programlisting>This example runs the <emphasis>demo.main</emphasis> ECL code
+          in the <emphasis>version1</emphasis> branch of the
+          <emphasis>gituser</emphasis>/<emphasis>gch-ecldemo-d</emphasis>
+          repository.</para>
+
+          <para>This feature is supported on both the ecl and eclcc command
+          line tools.</para>
```
Comment (JamesDeFabia) [inline docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml:302]: This feature is supported **by** both the ecl and eclcc command line tools.

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]
One minor comment inline

**Review by JamesDeFabia — APPROVED** [review]
Good from my POV

**Review by ghalliday — APPROVED** [review]
@g-pan please squash

**Pull request merged**